frontend/packages/schema-model/src/lib/mutations/ui-schema-reducers.ts

```diff
@@ -268,13 +269,34 @@ export const setPropertyName: UiSchemaReducer<SetPropertyNameArgs> =
     return renameNodePointer(newSchema, nodeToRename.pointer, newPointer);
   };
 
-export const toggleArrayField: UiSchemaReducer<string> =
-  (uiSchema, pointer) => {
-    const newSchema = deepCopy(uiSchema);
-    const node = getNodeByPointer(newSchema, pointer);
-    node.isArray = !node.isArray;
-    return newSchema;
-  };
+
+// update the "children" pointers 
+//include items in the pointer path if isArray is toggled on, or
+// remove items from the pointer path if isArray is toggled off. 
+// if isArray is toggled on, add items to the pointer path
+export const toggleArrayField: UiSchemaReducer<string> = (uiSchema, pointer) => {
+  const newSchema = deepCopy(uiSchema);
+  const node = getNodeByPointer(newSchema, pointer);
+  node.isArray = !node.isArray;
+
+  if (node.isArray) {
+    node.children.forEach((child) => {
+      const childNode = getNodeByPointer(newSchema, child);
+      childNode.pointer.replace(childNode.pointer, makePointer(
+        childNode.pointer, Keyword.Items));
+    });
+  } else {
+    node.children.forEach((child) => {
+      const childNode = getNodeByPointer(newSchema, child);
```

Comment:
Here you basically need to do the opposite of what is done in the `if`-block, i.e. go from
`#/properties/parentNode/items/properties/childNode`
to
`#/properties/parentNode/properties/childNode`,

i.e. replace `<parentPointer>/items` with `<parentPointer>`.